backend-go/nats.go

```diff
@@ -0,0 +1,64 @@
+package telestion
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/nats-io/nats.go"
+)
+
+// Initialize the NATS connection and register the health check subscriber.
+func initializeNats(config *Config, customNc *nats.Conn, customNcJson *nats.EncodedConn) (*nats.Conn, *nats.EncodedConn, error) {
+	var (
+		nc     *nats.Conn
+		ncJson *nats.EncodedConn
+		err    error
+	)
+
+	if customNc != nil {
+		if customNcJson == nil {
+			return nil, nil, errors.New("custom encoded nats connection is not defined but custom nats connection is (nc != nil && ncJson == nil)")
+		}
+
+		nc = customNc
+		ncJson = customNcJson
+	} else {
+		nc, ncJson, err = connectNats(config)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	// setup health check
+	_, _ = nc.Subscribe("__telestion__/health", func(msg *nats.Msg) {
+		response, _ := json.Marshal(healthcheck{0, config.ServiceName})
+		_ = msg.Respond(response)
+	})
```

Comment:
Waiting for formal service behavior specification => ok with me.

Reply:
I've created a new YouTrack issue to track this functionality:
https://telestion.youtrack.cloud/issue/TELESTION-470